core/app/models/spree/promotion_handler/promotion_batch_duplicator.rb

```diff
@@ -0,0 +1,37 @@
+module Spree
+  module PromotionHandler
+    class PromotionBatchDuplicator < Spree::PromotionDuplicatorCore
+      def initialize(promotion, promotion_batch_id, random_string: nil, code: nil)
+        @promotion = promotion
+        @promotion_batch_id = promotion_batch_id
+        @random_string = random_string || generate_random_string(4)
```

Comment:
Similarly to my second comment:
```suggestion
      def initialize(promotion, promotion_batch_id, random_string: generate_random_string(4), code: nil)
        @promotion = promotion
        @promotion_batch_id = promotion_batch_id
        @random_string = random_string
```

Reply:
Changed.